TwitchDownloaderCore/PlatformHelper.cs

```diff
@@ -0,0 +1,230 @@
+
+            try
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                {
+                    SetDirectoryPermissions(path);
+                }
+            }
+            catch { }
+
+            return directoryInfo;
+        }
+
+        public static void SetDirectoryPermissions(string path)
+        {
+            var folderInfo = new Mono.Unix.UnixFileInfo(path);
+            folderInfo.FileAccessPermissions = Mono.Unix.FileAccessPermissions.AllPermissions;
+            folderInfo.Refresh();
+        }
+
+        public static async Task<byte[]> GetImage(string cachePath, string imageUrl, string imageId, string imageScale, string imageType, CancellationToken cancellationToken = new())
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            byte[] imageBytes = null;
+
+            if (!Directory.Exists(cachePath))
+                PlatformHelper.CreateDirectory(cachePath);
+
+            string filePath = Path.Combine(cachePath!, $"{imageId}_{imageScale}.{imageType}");
+            if (File.Exists(filePath))
+            {
+                try
+                {
+                    await using var stream = File.Open(filePath, FileMode.Open, FileAccess.Read, FileShare.Read);
+                    byte[] bytes = new byte[stream.Length];
+                    stream.Seek(0, SeekOrigin.Begin);
+                    _ = await stream.ReadAsync(bytes, cancellationToken);
+
+                    //Check if image file is not corrupt
+                    if (bytes.Length > 0)
+                    {
+                        using SKImage image = SKImage.FromEncodedData(bytes);
+                        if (image != null)
+                        {
+                            imageBytes = bytes;
+                        }
+                        else
+                        {
+                            //Try to delete the corrupted image
+                            try
+                            {
+                                await stream.DisposeAsync();
+                                File.Delete(filePath);
+                            }
+                            catch { }
+                        }
+                    }
+                }
+                catch (IOException)
+                {
+                    //File being written to by parallel process? Maybe. Can just fallback to HTTP request.
+                }
+            }
+
+            // If fetching from cache failed
+            if (imageBytes != null)
+                return imageBytes;
+
+            // Fallback to HTTP request
+            if (imageUrl.Contains("kick.com"))
+            {
+                imageBytes = CurlImpersonate.GetCurlResponseBytes(imageUrl);
+            }
+            else
+            {
+                imageBytes = await HttpClient.GetByteArrayAsync(imageUrl, cancellationToken);
+            }
```

Comment:
This works for me without using curl to make the request. Did I just get lucky or did I just not download enough kick emotes?

Reply:
Not gonna bother testing, but this probably doesn't work without curl anymore since it seems that just about everything from kick requires using curl now.